OsmAnd/src/net/osmand/plus/plugins/odb/VehicleMetricsPlugin.kt

```diff
@@ -418,52 +416,40 @@ class VehicleMetricsPlugin(app: OsmandApplication) : OsmandPlugin(app),
 
 	@SuppressLint("MissingPermission")
 	fun connectToObd(activity: Activity, deviceInfo: BTDeviceInfo): Boolean {
-		if (currentConnectingState == OBDConnectionState.DISCONNECTED) {
-			if (connectedDeviceInfo == null) {
-				if (BLEUtils.isBLEEnabled(activity)) {
-					if (AndroidUtils.hasBLEPermission(activity)) {
-						onConnecting(deviceInfo)
-						if (socket != null && socket?.isConnected == true) {
-							socket?.close()
-							socket = null
-						}
-						OBDDispatcher.useInfoLogging =
-							PluginsHelper.getPlugin(OsmandDevelopmentPlugin::class.java)?.isEnabled == true
-
-						if (settings.SIMULATE_OBD_DATA.get() && deviceInfo.address.isEmpty()) {
-							onDeviceConnected(deviceInfo)
-							val simulator = ODBSimulationSource()
-							val input = simulator.reader
-							val output = simulator.writer
-							OBDDispatcher.setReadWriteStreams(input, output)
-							return true
-						}
+		if (currentConnectingState != OBDConnectionState.DISCONNECTED) {
+			disconnect()
```

Comment:
Not part of this request but connect disconnect should be moved to the thread of OBDDispatcher, so they can work asyncrhonously and all operations with socket will happen in 1 thread - in a thread of OBDDispatcher